src/app/modules/shared/models/entry.model.ts

```diff
@@ -9,6 +9,8 @@ export interface Entry {
   project_id?: string;
   owner_email?: string;
   description?: string;
+  customer_id?: string;
```

Comment:
When are these fields going to be null or undefined?

Reply:
When you are creating an entry.

Reply:
The time_entry table has the project id information, the customer info is not saved as part of it.

Reply:
True fact. However, from my perspective, the **time_entry** table is encapsulated into the API. The frontend does not ( and should not ) know about the database data structures. The frontend does ( and should ) know only about the data structures the endpoints are using ( which might be different from the database structures). In this case, the [Entry](https://github.com/ioet/time-tracker-ui/blob/master/src/app/modules/shared/models/entry.model.ts#L1) model/data structure/DTO represents the payload received from the `/time-entries` endpoint. Therefore, the customer and project information is not going to be empty since there is no way to create an entry without specifying the project ( and thus the customer).

Reply:
In addition, I think we can improve this data structure using compound objects. Something like this:
```
interface Entry {
  id: string;
  running: boolean;
  start_date: Date;
  end_date?: Date;
  activity_id?: string;
  technologies: string[];
  uri?: string;
  project :{
    id: String
    name: String
   customer:{
      id: String
      name: String
  },
  ...... 
}
```

Reply:
Yup, good idea